pdc/apps/release/views.py

```diff
@@ -1013,3 +1014,134 @@ def list(self, request, *args, **kwargs):
         %(SERIALIZER)s
         """
         return super(ReleaseVariantTypeViewSet, self).list(request, *args, **kwargs)
+
+
+class ReleaseGroupTypeViewSet(StrictQueryParamMixin,
+                              mixins.ListModelMixin,
+                              viewsets.GenericViewSet):
+    """
+    API endpoint that allows release-group types to be viewed.
+    """
+    serializer_class = ReleaseGroupTypeSerializer
+    queryset = models.ReleaseGroupType.objects.all()
+
+    def list(self, *args, **kwargs):
+        """
+        __Method__: GET
+
+        __URL__: $LINK:releasegrouptypes-list$
+
+        __Response__:
+
+        %(SERIALIZER)s
+        """
+        return super(ReleaseGroupTypeViewSet, self).list(*args, **kwargs)
+
+
+class ReleaseGroupsViewSet(ChangeSetModelMixin,
+                           StrictQueryParamMixin,
+                           viewsets.GenericViewSet):
+    """
+    API endpoint that allows release_group_types to be viewed or edited.
+    """
+
+    queryset = models.ReleaseGroup.objects.all()
+    serializer_class = ReleaseGroupSerializer
+    lookup_field = 'name'
+    lookup_value_regex = '[^/]+'
+    filter_class = filters.ReleaseGroupFilter
+
+    def create(self, request, *args, **kwargs):
+        """
+        __Method__: POST
+
+        __URL__: $LINK:releasegroups-list$
+
+        __Data__:
+
+        %(WRITABLE_SERIALIZER)s
+
+        __Response__:
+
+        %(SERIALIZER)s
+        """
+        data = request.data
+        keys = set(['description', 'name', 'type'])
+        arg_filter_map = ['active', 'releases']
+        allowed_keys = list(keys) + arg_filter_map
+        extra_keys = set(data.keys()) - set(allowed_keys)
+        if extra_keys:
+            return Response({'detail': '%s keys are not allowed' % list(extra_keys)},
+                            status=status.HTTP_400_BAD_REQUEST)
+        if 'releases' in data:
+            for key in data['releases']:
+                try:
+                    models.Release.objects.get(release_id=key)
+                except models.Release.DoesNotExist:
+                    return Response({'detail': '%s release does not exist' % key},
+                                    status=status.HTTP_404_NOT_FOUND)
+
```

Comment:
@ycheng-aa ; yes, which can verify the logic in serializer, but the error info is not usable, I think QE prefer to get the simple and clear error info from API.